FlowCryptCommon/Extensions/StringExtensions.swift

```diff
@@ -45,6 +45,20 @@ public extension String {
             }
         }
     }
+    
+    var fileExtension: String? {
+        guard let fileExtension = self.split(separator: ".").last else {
+            return nil
+        }
+        return String(fileExtension)
+    }
+    
+    func dropExtension() -> String {
+        guard let fileExtension = self.fileExtension else {
+            return self
+        }
+        return self.replacingOccurrences(of: ".\(fileExtension)", with: "")
+    }
```

Comment:
File named `made.by.pgpainless.pgp` would come out as `made.byainless` which is wrong. User a swift method to remove string suffix if there is one. Else a regex with a clear $ at the end.

Reply:
done